### PR TITLE
Solver.poisson support pre-integrated vfuncs

### PR DESCRIPTION
**Pull request by magnesium2400**
- `Solver.poisson` always multiplies the input vfunc `h` by `self.mass` to generate an integrated map
- This creates difficulties when solving for different maps, where some are integrated, and some are not: a new Solver object has to be created and its mass set to the identity
- It is easier to create a flag where the user can specify if the map is already integrated or not. This is what I have done here. 
- This also simplifies the code in `diffgeo.compute_geodesic_f` etc. 
- The default behaviour is preserved
- The figures in `Test_TriaMesh_Geodesics.ipynb` still look good when `vf = fem.poisson(Bi * divx)` is replaced by `vf = fem.poisson(divx, integrate=False)`

### CODE REVIEW

**Comment by m-reuter** [comment]
Hi @magnesium2400 thanks for this PR. I like to integrate it, but first the comments above need to be resolved. Fixing the doc-string to numpy format should also fix the doc build CI error.

**Comment by magnesium2400** [comment]
ok thanks @m-reuter ,  i will only have time in a few weeks but will do it then. and sorry, I couldnt find the tests anywhere (I could only see the notebooks) -- where/how should i write the tests (can I use pytest)? cheers.

**Comment by m-reuter** [comment]
Hi, tests are under lapy/utils/tests ( we will move them at some point into the project root, but currently you can add things there, if you find a similar tests already, if not, don't worry too much about the test). The other comments are more important.

File: lapy/solver.py [hunk 1]
```diff
@@ -808,6 +809,11 @@ def poisson(
             Neumann boundary condition as a tuple containing the index and
             data arrays of same length. The default, an empty tuple,
             corresponds to Neumann on all boundaries.
+        integrate: bool, default=True
+            Whether to integrate the right hand side over the surface using 
+            the mass matrix. If True, the right hand side is effectively 
+            replaced by ``B h``. If False, the right hand side is used as is, 
+            which corresponds to ``A x = h``.
```
[the remark this answers is not in the document]
Comment (magnesium2400) [reply lapy/solver.py]: updated docstring, replaced `integrate: bool` with `integrate : bool`

File: lapy/solver.py [hunk 2]
```diff
@@ -904,8 +910,9 @@ def poisson(
                 (dim, n_rhs), dtype=dtype
             )
         # compute right hand side
-        mass = self.mass.astype(dtype, copy=False)
-        b = mass * (h - nvec)
+        b = h - nvec
+        if integrate:
+            b = self.mass.astype(dtype, copy=False) * b
```
[the remark this answers is not in the document]
Comment (magnesium2400) [reply lapy/solver.py:917]: added test (not with boundary conditions)

File: lapy/diffgeo.py [hunk 3]
```diff
@@ -146,21 +146,21 @@ def compute_geodesic_f(
 
     gradf = compute_gradient(geom, vfunc)
     fem = Solver(geom, lump=True, use_cholmod=use_cholmod)
-    fem.mass = sparse.eye(fem.stiffness.shape[0], dtype=fem.stiffness.dtype)
+    # fem.mass = sparse.eye(fem.stiffness.shape[0], dtype=fem.stiffness.dtype)
```
[the remark this answers is not in the document]
Comment (magnesium2400) [reply lapy/diffgeo.py]: removed dead code comment

File: lapy/diffgeo.py [hunk 4]
```diff
@@ -200,21 +200,21 @@ def tria_compute_geodesic_f(
     fem = Solver(tria, lump=True, use_cholmod=use_cholmod)
     # div is the integrated divergence (so it is already B*div);
     # pass identity instead of B here
-    fem.mass = sparse.eye(fem.stiffness.shape[0])
+    # fem.mass = sparse.eye(fem.stiffness.shape[0])
```
[the remark this answers is not in the document]
Comment (magnesium2400) [reply lapy/diffgeo.py]: removed dead code comment

File: lapy/diffgeo.py [hunk 5]
```diff
@@ -503,20 +503,20 @@ def tria_compute_rotated_f(
     gradf = tria_compute_gradient(tria, vfunc)
     tn = tria.tria_normals()
     fem = Solver(tria, lump=True, use_cholmod=use_cholmod)
-    fem.mass = sparse.eye(fem.stiffness.shape[0], dtype=vfunc.dtype)
+    # fem.mass = sparse.eye(fem.stiffness.shape[0], dtype=vfunc.dtype)
```
[the remark this answers is not in the document]
Comment (magnesium2400) [reply lapy/diffgeo.py]: removed dead code comment

**Comment by magnesium2400** [comment]
thanks @m-reuter , i think i did all the changes that were brought up earlier. they were just small changes to the comments/docstrings. I added a simple test.

**Comment by m-reuter** [comment]
@copilot fix the ruff error

File: lapy/utils/tests/test_solver.py [hunk 6]
```diff
@@ -1,6 +1,7 @@
 """Tests for Solver.eigs and Solver.poisson — parameters and 2-D rhs support."""
 
 import numpy as np
+from scipy import sparse
 import pytest
```
Comment (m-reuter) [inline lapy/utils/tests/test_solver.py]:
```suggestion
import pytest
from scipy import sparse
```

**Comment by m-reuter** [comment]
Thanks for your contribution!
